firmware/fsm_msg_lisk.h

```diff
@@ -0,0 +1,127 @@
+void fsm_msgLiskGetAddress(LiskGetAddress *msg)
+{
+    CHECK_INITIALIZED
+
+    CHECK_PIN
+
+    RESP_INIT(LiskAddress);
+
+    HDNode *node = fsm_getDerivedNode(ED25519_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) return;
+
+    resp->has_address = true;
+
+    if (!hdnode_get_lisk_address(node, resp->address))
+		return;
+
+	if (msg->has_show_display && msg->show_display) {
+		char desc[16];
+		strlcpy(desc, "Address:", sizeof(desc));
+
+		if (!fsm_layoutAddress(resp->address, desc, true, 0, msg->address_n, msg->address_n_count)) {
+			return;
+		}
+	}
+
+    msg_write(MessageType_MessageType_LiskAddress, resp);
+
+    layoutHome();
+}
+
+void fsm_msgLiskGetPublicKey(LiskGetPublicKey *msg)
+{
+    CHECK_INITIALIZED
+
+    CHECK_PIN
+
+    RESP_INIT(LiskPublicKey);
+
+    HDNode *node = fsm_getDerivedNode(ED25519_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) return;
+
+    hdnode_fill_public_key(node);
+
+    resp->has_public_key = true;
+    resp->public_key.size = 32;
+
+	if (msg->has_show_display && msg->show_display) {
+		layoutLiskPublicKey(&node->public_key[1]);
+		if (!protectButton(ButtonRequestType_ButtonRequest_PublicKey, true)) {
+			fsm_sendFailure(FailureType_Failure_ActionCancelled, NULL);
+			layoutHome();
+			return;
+		}
+	}
+
+    memcpy(&resp->public_key.bytes, &node->public_key[1], sizeof(resp->public_key.bytes));
+
+    msg_write(MessageType_MessageType_LiskPublicKey, resp);
+
+    layoutHome();
+}
+
+void fsm_msgLiskSignMessage(LiskSignMessage *msg)
+{
+    CHECK_INITIALIZED
+
+    CHECK_PIN
+
+    RESP_INIT(LiskMessageSignature);
+
+    HDNode *node = fsm_getDerivedNode(ED25519_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) return;
+
+    hdnode_fill_public_key(node);
+
+    lisk_sign_message(node, msg, resp);
```

Comment:
Slightly strange behavior here compared to the rest of the Trezor codebase: if the user cancels signing, you get a Failure::ActionCancelled, as well as a LiskMessageSignature without the signature populated. Most everywhere else, you'd only get the failure after the user cancels.

IMO lisk_sign_message shouldn't be touching fsm, and instead that confirmation / cancelling should happen here.